File: 189-rotate-array/189-rotate-array.c
void rotate(int* nums, int numsSize, int k){
       
    int* temp=(int*)malloc(numsSize*sizeof(int));
    int j=0;
    
    if(k>numsSize)
        k=k%numsSize;
    
    for(int i=numsSize-k; i<numsSize; i++){

        temp[j++]=nums[i];
    }

    for (int i=0; i<numsSize-k; i++){

        temp[j++]=nums[i];
    }

    for(int i=0 ;i<numsSize; i++){

        nums[i]=temp[i];      
    }

    free(temp);
  
}

File: 189-rotate-array/189-rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        temp=nums.copy()
        
      
        for i in range(len(nums)):
            
            idx=(i+k)%len(nums)
            
            nums[idx]=temp[i]

        